src/foam/demos/examples/services.fbe

```diff
@@ -0,0 +1,436 @@
+    return taskId;
+  }
+  
+  @Override
+  public Task getTask(String taskId) {
+    Task task = taskMap.get(taskId);
+    getLogger().info("Task retrieved: " + taskId + " " + (task != null ? "found" : "not found"));
+    return task;
+  }
+  
+  @Override
+  public boolean deleteTask(String taskId) {
+    boolean existed = taskMap.containsKey(taskId);
+    taskMap.remove(taskId);
+    getLogger().info("Task deleted: " + taskId + " " + (existed ? "success" : "not found"));
+    return existed;
+  }
+}
+*/
+
+log('Separate Java implementation example shown above');
+
+##  Service Registration
+
+Register services in services.jrl using CSpec (Component Specification):
+--
+// Example services.jrl entry:
+/*
+p({
+  "class": "foam.core.boot.CSpec",
+  "name": "taskService",
+  "serve": true,
+  "authenticate": false,
+  "boxClass": "foam.demo.service.TaskServiceSkeleton",
+  "serviceScript": "return new foam.demo.service.ServerTaskService(x);",
+  "client": """
+    {
+      "class": "foam.box.SessionClientBox",
+      "delegate": {
+        "class": "foam.box.HTTPBox",
+        "url": "service/taskService"
+      }
+    }
+  """
+})
+*/
+
+log('Service registration example shown above');
+
+##  Client Usage
+
+Import and use services via context injection:
+--
+foam.CLASS({
+  package: 'foam.demo.service',
+  name: 'TaskManager',
+  
+  imports: [
+    'taskService?'  // Optional import with ? suffix
+  ],
+  
+  methods: [
+    {
+      name: 'manageTasks',
+      code: async function() {
+        if (!this.taskService) {
+          log('TaskService not available');
+          return;
+        }
+        
+        try {
+          // Create a task
+          var taskId = await this.taskService.createTask('Demo Task', 'Example description');
+          log('Created task:', taskId);
+          
+          // Retrieve the task
+          var task = await this.taskService.getTask(taskId);
+          log('Retrieved task:', task.title);
+          
+          // Delete the task
+          var deleted = await this.taskService.deleteTask(taskId);
+          log('Task deleted:', deleted);
+          
+        } catch (error) {
+          log('Service error:', error.message);
+        }
+      }
+    }
+  ]
+});
+
+// Create and use task manager
+var manager = foam.demo.service.TaskManager.create();
+// Note: In real usage, taskService would be injected via context
+log('TaskManager created - would use injected taskService in practice');
+
+##  Advanced Service Patterns
+
+##   Service Decorators
+
+Services can be decorated for additional functionality:
+--
+foam.CLASS({
+  package: 'foam.demo.service',
+  name: 'LoggingTaskServiceDecorator',
+  
+  implements: ['foam.demo.service.TaskService'],
+  
+  properties: [
+    {
+      class: 'Object',
+      name: 'delegate'
+    }
+  ],
+  
+  methods: [
+    {
+      name: 'createTask',
+      code: async function(title, description) {
+        log('LoggingDecorator: Creating task -', title);
```

Comment:
I'm thinking a service example with create, delete, and logging is perhaps not the best example, as it is effectively reimplementing the DAO/CRUD features FOAM provides out of the box for a Model.   If people saw this first, they might think this is how CRUD should be handled in FOAM.